track.go

```diff
@@ -200,7 +200,16 @@ func (track *baseTrack) bind(ctx webrtc.TrackLocalContext, specializedTrack Trac
 			encodedReader.Close()
 
 			// When there's another call to unbind, it won't block since we remove the current ctx from active connections
-			track.removeActivePeerConnection(ctx.ID())
+			ch := track.removeActivePeerConnection(ctx.ID())
+			if ch == nil {
+				// Its possible for unbind to be waiting to push into channel already
+				select {
+				case unbindCh := <-signalCh:
```

Comment:
so this will only hit if signalCh is blocking (waiting), which seems like will never happen, because signalCh emptys into `doneCh` then immediately returns. 
So there would need to be two instances of `<-signalCh` waiting?

Overall, using `close(channel)` is typically a bad idea for control flow unless its a a deliberate block
```
//goes to nowhere for deliberate blocking
<-someCh
```
or ending a ranged channel
```
for thing := range someCh {
//   this goes on for ever until 'close(someCh)' happens
}
```

I don't  think this will work as you intend